source/ed/connectors/fare_parser.cpp

```diff
@@ -101,6 +112,155 @@ void fare_parser::load_transitions() {
     }
 }
 
+bool fare_parser::is_valid(const navitia::fare::State& state) {
+    if (!state.mode.empty()) {
+        bool found = false;
+        for (const auto& mode : data.physical_modes) {
```

Comment:
You may probably shorten those checks using `contains()` https://github.com/CanalTP/utils/blob/master/functions.h#L217
And speedup using `data.physical_modes_map`.

Reply:
physical_modes_map is not present in ed::Data (the "data" available here)